### PR TITLE
test(parental-leave): Happy path

### PR DESCRIPTION
**Pull request by pshomov**
# ...

Parental leave application System E2E test. A few extra infrastructure items created while doing this:

- Dynamically generate emails using ethereal.mail service and registering those with SES to allow we can send email to those
- A script to load a subset of the `dev` env variables to be able to simulate locally configuration that is closer to reality. Most importantly, these are X-Road configuration settings.
- Extra logging for better troubleshooting

Slight changes were made to the application system UI to allow for an association(`aria-labelledby`) in a general way of titles with sections of the application. Please see the feature deployment to see the effects of the changes in action.

## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/system-e2e/src/fixtures/applications/users.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+[
+  {
+    "name": "Gervimaður Afríka",
+    "mobile": "0103019",
```
Comment (busla) [inline apps/system-e2e/src/fixtures/applications/users.json:4]:
```suggestion
    "phoneNumber": "0103019",
```

File: apps/system-e2e/src/fixtures/applications/users.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    "nationalId": "0101303019",
+    "role": "user"
+  },
+  {
+    "name": "Gervimaður Ameríku",
+    "mobile": "0102989",
```
Comment (busla) [inline apps/system-e2e/src/fixtures/applications/users.json:10]:
```suggestion
    "phoneNumber": "0102989",
```

File: apps/system-e2e/src/integration/applications/parental-leave.spec.ts [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+    })
+    cy.task('createEmailAccount', employer).then((email) => {
+      expect(email).to.be.a('string')
+      employerEmail = email as string
+    })
+  })
+
+  beforeEach(() => {
+    cy.idsLogin({
+      phoneNumber: fakeUser.mobile,
```
Comment (busla) [inline apps/system-e2e/src/integration/applications/parental-leave.spec.ts:34]:
```suggestion
      phoneNumber: fakeUser.phoneNumber,
```

**Review by busla — CHANGES_REQUESTED** [review]
we refactored all references from `mobile` to `phoneNumber` recently to  keep consistency.

File: apps/system-e2e/src/fixtures/applications/users.json (hunk 1 above)
Comment (pshomov) [reply apps/system-e2e/src/fixtures/applications/users.json:4]: Not seeing it on main, so will delay that until it lands there I suppose.

File: apps/system-e2e/src/fixtures/applications/users.json (hunk 2 above)
Comment (pshomov) [reply apps/system-e2e/src/fixtures/applications/users.json:10]: ditto

File: apps/system-e2e/src/integration/applications/parental-leave.spec.ts (hunk 3 above)
Comment (pshomov) [reply apps/system-e2e/src/integration/applications/parental-leave.spec.ts:34]: ditto

File: apps/system-e2e/src/support/email-account.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+  } | null>
+  email: string
+}) {
+  const client = new SESClient({ region: 'eu-west-1' })
+  await client.send(
+    new VerifyEmailAddressCommand({ EmailAddress: emailAccount.email }),
+  )
+  const verifyMsg = await emailAccount.getLastEmail(4)
+  if (verifyMsg && verifyMsg.text) {
+    console.log(`Verify message is ${verifyMsg.subject}: ${verifyMsg.text}`)
```
Comment (darrikonn) [inline apps/system-e2e/src/support/email-account.ts:36]: Do we want the logger for this?

File: apps/system-e2e/src/support/email-account.ts [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    const verifyUrl = verifyMsg.text.match(/https:\/\/email-verification.+/)
+    if (!verifyUrl || verifyUrl.length != 1) {
```
Comment (darrikonn) [inline apps/system-e2e/src/support/email-account.ts:38]: !==

File: apps/system-e2e/src/support/email-account.ts (hunk 4 above)
Comment (darrikonn) [reply apps/system-e2e/src/support/email-account.ts:36]: This comment goes for the whole file

**Review by darrikonn — APPROVED** [review]

**Review by barabrian — APPROVED** [review]
LGTM (Reviewed files that concern me)

**Review by qdd1 — APPROVED** [review]

File: apps/system-e2e/src/support/email-account.ts (hunk 4 above)
Comment (pshomov) [reply apps/system-e2e/src/support/email-account.ts:36]: I think this is mostly for troubleshooting problems while developing on a local instance. Will switch to our logging lib if we want this in DD at some point.

**Review by Thorrik58 — APPROVED** [review]

File: apps/system-e2e/src/support/email-account.ts (hunk 5 above)
Comment (pshomov) [reply apps/system-e2e/src/support/email-account.ts]: fixed

**Review by eirikurn — APPROVED** [review]
Nice!

**Review by RunarVestmann — APPROVED** [review]

**Review by thordurhhh — APPROVED** [review]

File: apps/system-e2e/src/lib/i18n-messages.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+//
+// // Create the `intl` object
+import { createIntl, createIntlCache, MessageDescriptor } from '@formatjs/intl'
+
```
Comment (busla) [inline apps/system-e2e/src/lib/i18n-messages.ts:4]:
```suggestion
import { cypressError } from '../support/utils'
```

File: apps/system-e2e/src/lib/i18n-messages.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+const cache = createIntlCache()
+const intl = createIntl(
+  {
+    locale: 'is',
+    onError: (err) => {
+      console.error(err)
```
Comment (busla) [inline apps/system-e2e/src/lib/i18n-messages.ts:10]:
```suggestion
      cypressError.error(err)
```

File: apps/system-e2e/src/support/email-account.ts [hunk 8]
```diff
@@ -0,0 +1,154 @@
+// use Nodemailer to get an Ethereal email inbox
+// https://nodemailer.com/about/
+import { createTestAccount } from 'nodemailer'
+// used to check the email inbox
+import { connect } from 'imap-simple'
+// used to parse emails from the inbox
+// const simpleParser = require('mailparser').simpleParser
+import { simpleParser } from 'mailparser'
+import axios from 'axios'
+
```
Comment (busla) [inline apps/system-e2e/src/support/email-account.ts:10]:
```suggestion
import { cypressError } from './utils'
```

File: apps/system-e2e/src/support/email-account.ts [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+  const client = new SESClient({ region: 'eu-west-1' })
+  await client.send(
+    new VerifyEmailAddressCommand({ EmailAddress: emailAccount.email }),
+  )
+  const verifyMsg = await emailAccount.getLastEmail(4)
+  if (verifyMsg && verifyMsg.text) {
+    console.log(`Verify message is ${verifyMsg.subject}: ${verifyMsg.text}`)
+    const verifyUrl = verifyMsg.text.match(/https:\/\/email-verification.+/)
+    if (!verifyUrl || verifyUrl.length !== 1) {
+      throw new Error(
```
Comment (busla) [inline apps/system-e2e/src/support/email-account.ts:39]:
```suggestion
      cypressError(
```

File: apps/system-e2e/src/support/email-account.ts [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+        Identities: [emailAccount.email],
+      }),
+    )
+    if (
+      emailVerifiedStatus.VerificationAttributes &&
+      emailVerifiedStatus.VerificationAttributes[emailAccount.email][
+        'VerificationStatus'
+      ] !== 'Success'
+    ) {
+      throw new Error(`Email identity still not validated in AWS SES`)
```
Comment (busla) [inline apps/system-e2e/src/support/email-account.ts:58]:
```suggestion
      cypressError(`Email identity still not validated in AWS SES`)
```

File: apps/system-e2e/src/support/email-account.ts [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+    }
+  } else {
+    throw new Error('Verification message not found.')
```
Comment (busla) [inline apps/system-e2e/src/support/email-account.ts:61]:
```suggestion
    cypressError('Verification message not found.')
```

File: apps/system-e2e/src/support/email-account.ts [hunk 12]
```diff
@@ -0,0 +1,154 @@
...
+
+          // and returns the main fields
+          return {
+            subject: mail.subject,
+            text: mail.text,
+            html: mail.html,
+          }
+        }
+      } catch (e) {
+        console.error(e)
```
Comment (busla) [inline apps/system-e2e/src/support/email-account.ts:145]:
```suggestion
        cypressError(e)
```

File: apps/system-e2e/src/support/tasks/email.ts [hunk 13]
```diff
@@ -0,0 +1,25 @@
+import { UnwrapPromise } from 'next/dist/lib/coalesced-function'
+import { makeEmailAccount, registerEmailAddressWithSES } from '../email-account'
```
Comment (busla) [inline apps/system-e2e/src/support/tasks/email.ts:2]:
```suggestion
import { makeEmailAccount, registerEmailAddressWithSES } from '../email-account'
import { cypressError } from '../utils'
```

File: apps/system-e2e/src/support/tasks/email.ts [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+      await registerEmailAddressWithSES(emailAccount)
+      emailAccounts[name] = emailAccount
+      return emailAccount.email
+    } else {
+      return emailAccounts[name].email
+    }
+  },
+
+  getLastEmail: ({ name, retries }: { name: string; retries: number }) => {
+    if (!emailAccounts[name]) throw new Error(`Email user not created yet`)
```
Comment (busla) [inline apps/system-e2e/src/support/tasks/email.ts:20]:
```suggestion
    if (!emailAccounts[name]) cypressError(`Email user not created yet`)
```

**Review by busla — APPROVED** [review]
LGTM

minor suggestion about throwing errors

**Pull request merged**